GitUI/CommandsDialogs/BrowseDialog/GitStatusMonitor.cs

```diff
@@ -455,7 +490,7 @@ private void Update()
                         {
                             lock (_statusSequence)
                             {
-                                if (_commandIsRunning && !ModuleHasChanged() && !cancelToken.IsCancellationRequested)
+                                if (!ModuleHasChanged() && !cancelToken.IsCancellationRequested)
```

Comment:
What about the following?
Keep line 521; _commandIsRunning = false;

```suggestion
                                if (cancelToken.IsCancellationRequested)
                                {
                                return;
                                }
                                
                                if (!ModuleHasChanged())
```

Reply:
I haven't had the time yet to think about your comments in detail. But I have watched it for long time and think to know two things for sure:
* The GitStatusMonitor stops doing anything because of `_commandIsRunning is true` (only resolvable by switching repos).
* If we reach this `finally` block, `GitExecutable.ExecuteAsync()` _has_ finished or been cancelled, i.e. nothing is pending in GE.

What have I missed?

The `git.exe` might continue to run detached in the background. We do not need to take care, don't we?
(I started to send `Ctrl+C` and to terminate the process on timeout - unfinished.)

> Keep line 521; _commandIsRunning = false;

I understand your concern about the delay. I need to take time thinking about how to do it without mixing states.

Reply:
Suggestion in fa36ff1147b8273b338b46647ef7b3f792d731a9
(_commandIsRunning  is more closely related to the cancellation token).